File: test/language/array_types/cpp/PackedAutoArrayRemovedEnumItemTest.cpp
#include "gtest/gtest.h"

#include "array_types/packed_auto_array_removed_enum_item/PackedAutoArrayRemovedEnumItem.h"

#include "zserio/SerializeUtil.h"

namespace array_types
{
namespace packed_auto_array_removed_enum_item
{

class PackedAutoArrayRemovedEnumItemTest : public ::testing::Test
{
protected:
    static const std::string BLOB_NAME;
};

const std::string PackedAutoArrayRemovedEnumItemTest::BLOB_NAME =
        "language/array_types/packed_auto_array_removed_enum_item.blob";

TEST_F(PackedAutoArrayRemovedEnumItemTest, writeReadFile)
{
    PackedAutoArrayRemovedEnumItem packedAutoArrayRemovedEnumItem;
    auto& packedArray = packedAutoArrayRemovedEnumItem.getPackedArray();
    packedArray.push_back(Traffic::NONE);
    packedArray.push_back(Traffic::LIGHT);
    packedArray.push_back(Traffic::MID);

    zserio::serializeToFile(packedAutoArrayRemovedEnumItem, BLOB_NAME);
    auto readPackedAutoArrayRemovedEnumItem =
            zserio::deserializeFromFile<PackedAutoArrayRemovedEnumItem>(BLOB_NAME);
    ASSERT_EQ(packedAutoArrayRemovedEnumItem, readPackedAutoArrayRemovedEnumItem);
}

TEST_F(PackedAutoArrayRemovedEnumItemTest, writeRemovedException)
{
    PackedAutoArrayRemovedEnumItem packedAutoArrayRemovedEnumItem;
    auto& packedArray = packedAutoArrayRemovedEnumItem.getPackedArray();
    packedArray.push_back(Traffic::NONE);
    packedArray.push_back(Traffic::LIGHT);
    packedArray.push_back(Traffic::MID);
    packedArray.push_back(Traffic::ZSERIO_REMOVED_HEAVY);

    ASSERT_THROW(zserio::serialize(packedAutoArrayRemovedEnumItem), zserio::CppRuntimeException);
}

} // namespace packed_auto_array_removed_enum_item
} // namespace array_types

File: test/language/array_types/java/array_types/PackedAutoArrayRemovedEnumItemTest.java
package array_types;

import static org.junit.jupiter.api.Assertions.*;
import org.junit.jupiter.api.Test;

import array_types.packed_auto_array_removed_enum_item.PackedAutoArrayRemovedEnumItem;
import array_types.packed_auto_array_removed_enum_item.Traffic;

import zserio.runtime.ZserioError;
import zserio.runtime.io.BitBuffer;
import zserio.runtime.io.SerializeUtil;

public class PackedAutoArrayRemovedEnumItemTest
{
    @Test
    public void writeReadFile()
    {
        final PackedAutoArrayRemovedEnumItem packedAutoArrayRemovedEnumItem =
                new PackedAutoArrayRemovedEnumItem(new Traffic[] {Traffic.NONE, Traffic.LIGHT, Traffic.MID});

        SerializeUtil.serializeToFile(packedAutoArrayRemovedEnumItem, BLOB_NAME);

        final PackedAutoArrayRemovedEnumItem readPackedAutoArrayRemovedEnumItem =
                SerializeUtil.deserializeFromFile(PackedAutoArrayRemovedEnumItem.class, BLOB_NAME);
        assertEquals(packedAutoArrayRemovedEnumItem, readPackedAutoArrayRemovedEnumItem);
    }

    @Test
    public void writeRemovedException()
    {
        final PackedAutoArrayRemovedEnumItem packedAutoArrayRemovedEnumItem =
                new PackedAutoArrayRemovedEnumItem(new Traffic[] {
                        Traffic.NONE, Traffic.LIGHT, Traffic.MID, Traffic.ZSERIO_REMOVED_HEAVY});
        assertThrows(ZserioError.class, () -> SerializeUtil.serialize(packedAutoArrayRemovedEnumItem));
    }

    private static final String BLOB_NAME = "packed_auto_array_removed_enum_item.blob";
}
